### PR TITLE
Laboratory work #4, Zakharova Sofja - 21FPL2

### PR DESCRIPTION
**Pull request by solonarr**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_fill_words_by_ngrams/main.py [hunk 1]
```diff
@@ -98,6 +146,45 @@ def run(self, seq_len: int, prompt: str) -> str:  # type: ignore
...
+        encoded_prompt = self._word_processor.encode(prompt)
+
+        if not encoded_prompt:
+            raise ValueError
+
+        while seq_len >= 1:
+            next_word_dict = self._model.generate_next_token(encoded_prompt)
+
+            if next_word_dict is None:
+                raise ValueError
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:161]: .

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_fill_words_by_ngrams/main.py [hunk 2]
```diff
@@ -56,6 +80,27 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> str:  #
         Raises:
             ValueError: In case of inappropriate type input argument or if input argument is empty.
         """
+        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) == 0:
+            raise ValueError
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:84]: add message

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
fix previous comments

File: lab_4_fill_words_by_ngrams/main.py (hunk 1 above)
Comment (solonarr) [reply lab_4_fill_words_by_ngrams/main.py:161]: fixed

File: lab_4_fill_words_by_ngrams/main.py [hunk 3]
```diff
@@ -39,6 +58,11 @@ def _put(self, element: str) -> None:
         Raises:
             ValueError: In case of inappropriate type input argument or if input argument is empty.
         """
+        if not isinstance(element, str) or len(element) == 0:
+            raise ValueError
```
[the remark this answers is not in the document]
Comment (solonarr) [reply lab_4_fill_words_by_ngrams/main.py:62]: fixed

File: lab_4_fill_words_by_ngrams/main.py (hunk 2 above)
Comment (solonarr) [reply lab_4_fill_words_by_ngrams/main.py:84]: fixed

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**